### ISSUE TITLE
Tween runtime and tell are broken with limited use

### ISSUE THREAD

**Issue by ricochet1k** [godotengine/godot#26988]
**Godot version:** v3.1.rc3.official

**Issue description:**
I would like to use Tween to animate a bunch of sprites, and I want to allow these animations to chain arbitrarily, and also be able to wait for all the tweens to finish. I am struggling to get any of these to work properly.

Here's a log dump from the reproduction project below, with some comments added:
```
time:0.00  active:False  tell:0.00  runtime:0.00   _ready   # good
time:0.00  active:False  tell:0.00  runtime:0.50   before start   # good
time:0.00  active: True  tell:0.00  runtime:0.50   after start   # good
time:0.00  active: True  tell:0.08  runtime:0.50   tween_started: :position   # Note A
time:0.00  active: True  tell:0.08  runtime:0.50   tween_started: :callback_a
time:0.41  active: True  tell:0.41  runtime:0.50   tween_completed: :position   # Note B
time:0.41  active: True  tell:0.50  runtime:0.50   callback_a   # good
time:0.41  active: True  tell:0.50  runtime:0.50   callback_a after   # Note C
time:0.41  active: True  tell:0.00  runtime:0.50   tween_completed: :callback_a   # Note D
time:0.51  active: True  tell:0.09  runtime:0.30   tween_started: :rotation_degrees   # Same as Note A
time:0.51  active: True  tell:0.09  runtime:0.30   tween_started: :callback_b
time:0.70  active: True  tell:0.19  runtime:0.30   tween_completed: :rotation_degrees   # Same as note B
time:0.80  active: True  tell:0.30  runtime:0.30   callback_b
time:0.80  active: True  tell:0.30  runtime:0.30   callback_b after   # Note E
time:0.80  active: True  tell:0.00  runtime:0.30   tween_completed: :callback_b   # Same as note D
time:0.90  active: True  tell:0.10  runtime:2.00   tween_started: :scale
time:0.90  active: True  tell:0.10  runtime:2.00   tween_started: :callback_c
time:2.80  active: True  tell:1.90  runtime:2.00   tween_completed: :scale   # Same as note A
time:2.80  active: True  tell:0.00  runtime:2.00   tween_completed: :callback_c
time:2.90  active: True  tell:0.00  runtime:2.00   callback_c
```
For reference, the `position` and `callback_a` durations are 0.5, `rotation_degrees` duration is 0.25, `callback_b` duration is 0.3, `scale` and `callback_c` have a duration of 2, `callback_c` is deferred. This is run with a Fixed FPS of 10 for repeatability.

- A: It's sad that this has a tell of 0.08, since it was started at 0. Can anything be done about this?
- B: `position` completes at 0.41, even though its duration is 0.5, especially if you think it started at 0.08 like the `tell` on its `tween_started` signal reports. This seems just wrong, it hasn't run for long enough. I would expect it to complete on the next frame. Presumably it is completing because the next frame would be > 0.5?
- C: At this point, we have just completed 2 tweens with a duration of 0.5, and just added tweens with interpolations of 0.25 and 0.3, but `tell()` and `get_runtime()` don't report anything different. If at this point you want to know how long it will take for the current tweens to finish, you are out of luck.
- D: This is even more confusing. We have just completed 2 tweens of 0.5 duration, just added tweens of 0.25 and 0.3, yet tell and get_runtime paint the picture that we just started tweens of a maximum of 0.5?
- E: We just started 2 tweens of 2 seconds, but runtime doesn't show them.

What I would like to see:
1. Tween_started signals are sent before tell is incremented. At Note A, tell should be 0
2. Adding tweens immediately affects runtime. At the position of Note C above, I'd expect tell to still be 0.5 but runtime should be 0.5 + 0.3 = 0.8
3. Complete tweens on the exact frame that their duration passes. At Note B, time and tell should be > 0.5
4. Tell doesn't drop to 0 until after all the completed callbacks have passed, and runtime and active are reset at the same time. At Note D, runtime should be 0.3. At the end at callback_c, active should be False and runtime should be 0.
5. An all_tweens_completed() signal would be very helpful, signaling the moment that all tweens are complete, all callbacks are called, and active has just changed to False.

Do you agree with all the above? Any clever solutions I missed?

I might start trying to fix these issues if no one objects, unless someone else wants to tackle them.

**Steps to reproduce:**
Run the reproduction project, look at the log.

**Minimal reproduction project:**
[tween_broken.zip](https://github.com/godotengine/godot/files/2958880/tween_broken.zip)

**Comment by akien-mga** [comment]
Can you still reproduce this bug in Godot 3.2.1 or any later release (e.g. [3.2.2-rc2](https://godotengine.org/article/release-candidate-godot-3-2-2-rc-2))?

**Comment by ricochet1k** [comment]
Just tried on 3.2.3.rc4, it actually seems worse now. When `callback_a` is called it schedules `callback_b` and `callback_b` never happens.

```
time:0.00  active:False  tell:0.00  runtime:0.00   _ready
time:0.00  active:False  tell:0.00  runtime:0.50   before start
time:0.00  active: True  tell:0.00  runtime:0.50   after start
time:0.00  active: True  tell:0.02  runtime:0.50   tween_started: :position
time:0.00  active: True  tell:0.02  runtime:0.50   tween_started: :callback_a
time:0.47  active: True  tell:0.47  runtime:0.50   tween_completed: :position
time:0.47  active: True  tell:0.50  runtime:0.50   callback_a
time:0.47  active: True  tell:0.50  runtime:0.50   callback_a after
time:0.47  active: True  tell:0.00  runtime:0.50   tween_completed: :callback_a
```

**Comment by ricochet1k** [comment]
Just ran again on the latest 3.2 branch, looks like 2d42625184cd70e954fe6ef8dc6c95a23e13b9e2 brought this back to what it was before.

```
time:0.00  active:False  tell:0.00  runtime:0.00   _ready
time:0.00  active:False  tell:0.00  runtime:0.50   before start
time:0.00  active: True  tell:0.00  runtime:0.50   after start
time:0.00  active: True  tell:0.02  runtime:0.50   tween_started: :position
time:0.00  active: True  tell:0.02  runtime:0.50   tween_started: :callback_a
time:0.43  active: True  tell:0.43  runtime:0.50   tween_completed: :position
time:0.43  active: True  tell:0.50  runtime:0.50   callback_a
time:0.43  active: True  tell:0.50  runtime:0.50   callback_a after
time:0.43  active: True  tell:0.00  runtime:0.50   tween_completed: :callback_a
time:0.53  active: True  tell:0.10  runtime:0.30   tween_started: :rotation_degrees
time:0.53  active: True  tell:0.10  runtime:0.30   tween_started: :callback_b
time:0.73  active: True  tell:0.20  runtime:0.30   tween_completed: :rotation_degrees
time:0.83  active: True  tell:0.30  runtime:0.30   callback_b
time:0.83  active: True  tell:0.30  runtime:0.30   callback_b after
time:0.83  active: True  tell:0.00  runtime:0.30   tween_completed: :callback_b
time:0.93  active: True  tell:0.10  runtime:2.00   tween_started: :scale
time:0.93  active: True  tell:0.10  runtime:2.00   tween_started: :callback_c
time:2.83  active: True  tell:1.90  runtime:2.00   tween_completed: :scale
time:2.83  active: True  tell:0.00  runtime:2.00   tween_completed: :callback_c
time:2.93  active: True  tell:0.00  runtime:2.00   callback_c
```

**Comment by KoBeWi** [comment]
This seems to mention multiple issues, some of which are resolved already in 3.3, and most likely all of them are fixed with new Tweens in 4.0.
